lib/controllers/RelationalController.ts

```diff
@@ -420,15 +421,53 @@ export abstract class RelationalController extends CRUDController {
     }
     else {
 
-      await this.as(user).document.update(
+      await this.sdk.document.update(
         index,
         collection,
         document,
         body,
       );
     }
   }
-  
+
+  /**
+   * 
+   * @param index
+   * @param collection
+   * @param document
+    * @param nestedFields : contain field name in the document to get, collection name and index name of documents that are linked
+   */
+  protected async genericGet<TKDocumentContent extends KDocumentContent> (index: string, collection : string, documentId : string, nestedFields : FieldPath[] = []): Promise<KDocument<TKDocumentContent>> {
+    const document = await this.sdk.document.get<TKDocumentContent>(index, collection, documentId);
+    for (const nestedField of nestedFields) {
+      if (document._source[nestedField.field]) {
+        document._source[nestedField.field] = await this.getRequestRaw(nestedField.index, nestedField.collection, document._source[nestedField.field]);
+      }
+    }
+    return document;
+  }
+
+  private async getRequestRaw (index: string, collection : string, document : string) : Promise<KDocumentContent> {
```

Comment:
Why is this method async if nothing is awaited in it ?

Reply:
the return is async